src/main/scala/nl.knaw.dans.easy.bag2deposit/ddm/LicenseRewriteRule.scala

```diff
@@ -0,0 +1,121 @@
+    if (! isLicenseUri(node))
+      node
+    else {
+      val correctURI = getLicenseUri(supportedLicenses, variantToLicense, node)
+      <dct:license xsi:type="dct:URI">{ correctURI.toString }</dct:license>
+    }
+  }
+
+  private def isLicenseUri(node: Node): Boolean = {
+    if (!("license" == node.label) || !("http://purl.org/dc/terms/" == node.namespace) || !hasXsiType(node, "URI")) {
+      false
+    } else {
+      // validate it is a valid URI
+      Try(new URI(node.text.trim)).isSuccess
+    }
+  }
+
+  private def hasXsiType(node: Node, xsiType: String): Boolean = {
+    val attributes = node.attributes
+    if (attributes == null) return false
+    val binding = NamespaceBinding("xsi", "http://www.w3.org/2001/XMLSchema-instance", node.scope)
+    val maybeTypes = attributes.get("http://www.w3.org/2001/XMLSchema-instance", binding, "type")
+    if ( maybeTypes.isEmpty ) return false
+    val types = maybeTypes.get
+    for(item <- types){
+      val text = item.text
+      return xsiType == text || text.endsWith(":" + xsiType)
+    }
+    false
+
+  }
+  private def getLicenseUri(supportedLicenses: List[URI], variantToLicense: Map[String, String], licenseNode: Node): URI = {
+    val licenseText = Option(licenseNode)
+      .map(_.text)
+      .map(_.trim)
+      .map(removeTrailingSlash(_))
+      .map(s => variantToLicense.getOrElse(s,s))
+      .getOrElse(throw new IllegalArgumentException("License node is null"))
+
+    try {
+      if (!isLicenseUri(licenseNode))
+        throw new IllegalArgumentException("Not a valid license node")
+      var licenseUri = new URI(licenseText)
+      val licenseUriFinal = licenseUri
+      val maybeLicenseUri = normalizeScheme(supportedLicenses, licenseUri)
+      if(maybeLicenseUri.equals(Option.empty))
+        new IllegalArgumentException(String.format("Unsupported license: %s", licenseUriFinal))
+      licenseUri = maybeLicenseUri.get
+      supportedLicenses.filter(v => v== licenseUri).headOption.getOrElse(throw new IllegalArgumentException(String.format("Unsupported license: %s", licenseUri)))
+    } catch {
+      case e: Exception =>
+        logger.error("Invalid license URI: {}", licenseText, e)
+        throw new IllegalArgumentException("Not a valid license URI", e)
+    }
+  }
+
+  private def removeTrailingSlash(s: String): String = {
+    if (s.endsWith("/")) return s.substring(0, s.length - 1)
+    s
+  }
+
+  private def normalizeScheme(supportedLicenses: List[URI], uri: URI): Option[URI] = {
+    val schemes: Set[String] = Set("https", "http")
+    var result = Option.empty[URI]
+    for (license <- supportedLicenses) {
+      if (StringUtils.equals(license.getHost, uri.getHost)
+        && StringUtils.equals(license.getPath, uri.getPath)
+        && license.getPort == uri.getPort
+        && StringUtils.equals(license.getQuery, uri.getQuery)
+        && schemes.contains(uri.getScheme))
+        result = Option(license)
+    }
+    result
```

Comment:
in scala you don't need `equals` for `==`

```suggestion
    supportedLicenses.find { license =>
      license.getHost == uri.getHost &&
        license.getPath == uri.getPath &&
        license.getPort == uri.getPort &&
        license.getQuery == uri.getQuery &&
        schemes.contains(uri.getScheme)
    }
```